python/cudf/cudf/core/dataframe.py

```diff
@@ -217,6 +217,13 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
                         ),
                     )
                 )
+        elif hasattr(data, "__array_interface__") or hasattr(
+            data, "__cuda_array_interface__"
+        ):
+            new_df = self.from_records(data, index=index, columns=columns)
```

Comment:
Can you update the docstring on `from_records` to make it clear it accepts array-like objects in either GPU or host memory?

Reply:
Actually, nevermind this isn't the right function that a general array should go through: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.from_records.html

Reply:
Looks like `Pandas.DataFrame.from_records` isn't accepting 1-d arrays, instead passing them through `_from_columns` flow.

Reply:
I don't think `from_records` is the intended path for 2-d arrays either. It's specifically meant for structured / record arrays.

Reply:
Not using `from_records` currently and introduced a new function `_from_arrays` to handle array-like inputs to dataframe.